harness/determined/common/experimental/token.py

```diff
@@ -0,0 +1,109 @@
+import enum
+from typing import List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+
+
+class TokenType(enum.Enum):
+    # UNSPECIFIED is internal to the bound API and is not be exposed to the front end
+    USER_SESSION = bindings.v1TokenType.USER_SESSION.name
+    ACCESS_TOKEN = bindings.v1TokenType.ACCESS_TOKEN.name
+
+
+class AccessToken:
+    """
+    A class representing a AccessToken object that contains user session token info and
+    access token info.
+    It can be obtained from :func:`determined.experimental.client.list_access_tokens`
+    Attributes:
+        session: HTTP request session.
+        token_id: (int) The ID of the access token in user sessions table.
+        user_id: (int) Unique ID for the user.
+        expiry: (str) Timestamp expires at reported.
+        created_at: (str) Timestamp created at reported.
+        token_type: (TokenType) Token type of the token.
+        revoked: (Mutable, Optional[bool]) The datetime when the token was revoked.
+            Null if the token is still active.
+        description: (Mutable, Optional[str]) Human-friendly description of token.
+
+    Note:
+        Mutable properties may be changed by methods that update these values either automatically
+        (eg. `revoke_tokens`, `edit_tokens`) or explicitly with :meth:`reload()`.
+    """
+
+    def __init__(self, token_id: int, session: api.Session):
+        self.token_id = token_id
+        self._session = session
+
+        self.user_id: Optional[int] = None
+        self.expiry: Optional[str] = None
+        self.created_at: Optional[str] = None
+        self.token_type: Optional[TokenType] = None
+        self.revoked: Optional[bool] = None
+        self.description: Optional[str] = None
+
+    def _hydrate(self, tokenInfo: bindings.v1TokenInfo) -> None:
+        self.user_id = tokenInfo.userId
+        self.expiry = tokenInfo.expiry
+        self.created_at = tokenInfo.createdAt
+        self.token_type = tokenInfo.tokenType
+        self.revoked = tokenInfo.revoked if tokenInfo.revoked is not None else False
+        self.description = tokenInfo.description if tokenInfo.description is not None else ""
+
+    def reload(self) -> None:
+        resp = bindings.get_GetAccessTokens(
+            session=self._session, tokenIds=[self.id], showInactive=True
```

Comment:
does `self.id` exist? either change this to `self.token_id` or add a property to this class:

```
@property
    def id(self) -> int:
        return self._id
```